FrameworkSystem/private/standardLogging/gLogging.py

```diff
@@ -0,0 +1,370 @@
+
+  def _setChildrenLevel(self, level):
+    """
+    Set the level of the children if it is not modified by the user
+    """
+    if not self._levelModified:
+      self._level = level
+    for child in self._children.values():
+      child._setChildrenLevel(level)
+
+  @staticmethod
+  def getAllPossibleLevels():
+    """
+    :return: a list of all levels available
+    """
+    return LogLevels.getLevelNames()
+
+  def always(self, sMsg, sVarMsg=''):
+    """
+    Always level
+    """
+    level = LogLevels.getLevelValue('ALWAYS')
+    return self._createLogRecord(level, sMsg, sVarMsg)
+
+  def notice(self, sMsg, sVarMsg=''):
+    """
+    Notice level
+    """
+    level = LogLevels.getLevelValue('NOTICE')
+    return self._createLogRecord(level, sMsg, sVarMsg)
+
+  def info(self, sMsg, sVarMsg=''):
+    """
+    Info level
+    """
+    level = LogLevels.getLevelValue('INFO')
+    return self._createLogRecord(level, sMsg, sVarMsg)
+
+  def verbose(self, sMsg, sVarMsg=''):
+    """
+    Verbose level
+    """
+    level = LogLevels.getLevelValue('VERBOSE')
+    return self._createLogRecord(level, sMsg, sVarMsg)
+
+  def debug(self, sMsg, sVarMsg=''):
+    """
+    Debug level
+    """
+    level = LogLevels.getLevelValue('DEBUG')
+    return self._createLogRecord(level, sMsg, sVarMsg)
+
+  def warn(self, sMsg, sVarMsg=''):
+    """
+    Warn
+    """
+    level = LogLevels.getLevelValue('WARN')
+    return self._createLogRecord(level, sMsg, sVarMsg)
+
+  def error(self, sMsg, sVarMsg=''):
+    """
+    Error level
+    """
+    level = LogLevels.getLevelValue('ERROR')
+    return self._createLogRecord(level, sMsg, sVarMsg)
+
+  def exception(self, sMsg="", sVarMsg='', lException=False, lExcInfo=False):
+    """
+    Exception level
+    """
+    level = LogLevels.getLevelValue('ERROR')
+    return self._createLogRecord(level, sMsg, sVarMsg, exc_info=True)
+
+  def fatal(self, sMsg, sVarMsg=''):
+    """
+    Critical level
+    """
+    level = LogLevels.getLevelValue('FATAL')
+    return self._createLogRecord(level, sMsg, sVarMsg)
+
+  def _createLogRecord(self, level, sMsg, sVarMsg, exc_info=False):
+    """
+    Create a log record according to the level of the message. The log record is always sent to the different backends
+    Backends have their own levels and can manage the display of the message or not according to the level. 
+    Nevertheless, backends and the logger have the same level value, 
+    so we can test if the message will be displayed or not. 
+    :params level: positive integer representing the level of the log record
+    :params sMsg: string representing the message
+    :params sVarMsg: string representing an optional message
+    :params exc_info: boolean representing the stacktrace for the exception
+
+    :return: boolean representing the result of the log record creation
+    """
+    self._logger.log(level, "%s %s", sMsg, sVarMsg, exc_info=exc_info, extra={'componentname': self.getName()})
```

Comment:
I writed a comment on that
I think it is not a good idea because it corresponds to a Log Record attributes and they are not in camel case. What do you think ? 
I created a new extra attribute for the sVarMsg, so they are totally separated :-)